### PR TITLE
Single product  payment

### PR DESCRIPTION
**Pull request by kotesh-arya**
# Featues consisted in this PR
- Single Product Page
- Payment Integration using Razorpay
- React Toastify pop-ups for better User-Experience

### CODE REVIEW

**Comment by irajbhinde** [comment]
All the features are working well, only suggestion would be, after the payment is made, redirect the user to the order confirmation page where you can use the order ID received from Razorpay.

File: src/App.js [hunk 1]
```diff
@@ -9,9 +9,24 @@ import { UserInfo } from "./pages/UserInfo/UserInfo";
 import { Routes, Route } from "react-router-dom";
 import { RequiresAuth } from "./RequiresAuth";
 import { NotFound } from "./pages/NotFound/NotFound";
+import { ProductPage } from "../src/pages/ProductPage/ProductPage";
+import { ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+const notify = () => {
+  toast("🦄 Wow so easy!", {
```
Comment (Namrata-J) [inline src/App.js]: if you added it for testing purpose, and if not needed for future use, then you can remove this toast from here, and also the button at line no. 29 in this file.

PS: all your toasts used at other places are working as expected.

File: src/contexts/authContext.js [hunk 2]
```diff
@@ -44,12 +46,29 @@ const AuthProvider = ({ children }) => {
             navigate("/ProductList");
           }
 
-          localStorage.setItem("JWT_TOKEN", encodedToken);
           console.log(foundUser);
```
Comment (Namrata-J) [inline src/contexts/authContext.js]: remove all your console's, if not needed anymore!!

File: src/contexts/authContext.js [hunk 3]
```diff
@@ -73,19 +92,45 @@ const AuthProvider = ({ children }) => {
...
+            draggable: true,
+            progress: undefined,
+          });
         }
       }
     }
   };
   const logoutUser = () => {
     navigate("/");
     userDispatch({ type: "USER_LOGOUT" });
```
Comment (Namrata-J) [inline src/contexts/authContext.js]: suggestion: store the dispatch type in constants or variables, do not hardcode it.

File: src/index.css [hunk 4]
```diff
@@ -1,4 +1,4 @@
-@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");
+/* @import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap"); */
```
Comment (Namrata-J) [inline src/index.css]: remove commented code from everywhere

**Review by Namrata-J — APPROVED** [review]
rest of your code looks modular and and very well structured.

File: src/contexts/authContext.js [hunk 5]
```diff
@@ -73,19 +92,45 @@ const AuthProvider = ({ children }) => {
           });
           console.log(createdUser);
```
Comment (ashishkatiyar2404) [inline src/contexts/authContext.js]: you can remove this

File: src/contexts/authContext.js (hunk 3 above)
Comment (ashishkatiyar2404) [reply src/contexts/authContext.js]: yes, hardcoding dispatch type is not a good practise

File: src/pages/Cart/Cart.jsx [hunk 6]
```diff
@@ -1,18 +1,83 @@
...
+    }
+    const options = {
+      key: "rzp_test_I09NM8PpgOnIY7",
+      currency: "INR",
+      amount: amount * 100,
+      name: "Daily fashion INC",
+      Description: "Thanks for shopping",
+      handler: function (response) {
+        dispatch({ type: "PAYMENT_DONE" });
+        // alert(response.razorpay_payment_id);
```
Comment (ashishkatiyar2404) [inline src/pages/Cart/Cart.jsx:60]: you can remove commented code

File: src/pages/Cart/CartProduct.jsx [hunk 7]
```diff
@@ -64,11 +70,54 @@ function CartProduct({
...
-          <button className={`btn btn-primary ${CartCSS["cart-card-button"]}`}>
+          <button
+            className={`btn btn-primary ${CartCSS["cart-card-button"]}`}
+            onClick={() => {
+              if (wishlist.some((item) => item.id === id)) {
+                console.log("item exists");
+                // wishlistDispatch({
+                //   type: "PRE_EXISTED_WISHLIST_ITEM",
+                //   payload: title,
+                // });
```
Comment (ashishkatiyar2404) [inline src/pages/Cart/CartProduct.jsx]: here too

**Review by ashishkatiyar2404 — APPROVED** [review]
Suggestion; - Remove commented code if there is no use for it in the future.
Appreciation: It- Good to see you have taken care of the responsiveness of your project.

You can merge

File: src/App.js (hunk 1 above)
Comment (kotesh-arya) [reply src/App.js]: Thanks for pointing, removed it!


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not reusable

**Pull request merged**